feat(config/package-rules): `matchCurrentAge` (#23264)
renovatebot/renovate: docs/usage/configuration-options.md

}
```

### matchCurrentAge

> Use this field if you want to limit a `packageRule` to package with certain age.
> Eg. If you want to group packages which are older than 2 months ie. haven't been updated formore than 2 months

Author: ```suggestion
Use this field if you want to match packages that lie in a certain age range.

For Eg. If you want to group packages which are older than 2 months
```

}
```

### matchCurrentAge

> Use this field if you want to match packages that lie in a certain age range.

Reviewer: ```suggestion
Use this field if you want to match packages that satisfy the specified age range.
```

> For Eg. If you want to group packages which are older than 2 months

Reviewer 2: ```suggestion
For example, if you want to group packages which are older than 2 months:
```

```json
{
  "packageRules": [
    {
      "matchCurrentAge": "> 2 months",
      "groupName": "vital"
    }
  ]
}
```

> Invalid if used outside of a `packageRule`.

Reviewer 2: ```suggestion
```

We can drop this, you've set `parent = packageRules` in the code.

That should make it so this config option docs has a table row like this:

| Name | Value |
| ---- | ------ |
| parent | packageRules |

}
```

### matchCurrentAge

Use this field if you want to match packages that satisfy the specified age range.

For example, if you want to group packages which are older than 2 months:

```json
{
  "packageRules": [
    {

>       "matchCurrentAge": "> 2 months",

Reviewer: Need to document which word are supported - years, months, weeks, days?

      "groupName": "vital"
    }
  ]
}
```

> The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.

Reviewer: ```suggestion
The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.
Only one date part is supported, e.g. you _cannot_ do `> 1 year 1 month` and instead should do `> 13 months`.
```

}
```

### matchCurrentAge

Use this field if you want to match packages that satisfy the specified age range.

For example, if you want to group packages which are older than 2 months:

```json
{
  "packageRules": [
    {
      "matchCurrentAge": "> 2 months",
      "groupName": "vital"
    }
  ]
}
```

The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.

> Only one date part is supported, e.g. you _cannot_ do `> 1 year 1 month` and instead should do `> 13 months`.

Reviewer 2: ```suggestion
Only _one_ date part is supported, so you _cannot_ do `> 1 year 1 month`.
Instead you should do `> 13 months`.
```

<!-- prettier-ignore -->
!!! note

>     We recommend using these words hour(s), day(s), week(s), month(s) and year(s) to specify time formats.

Reviewer 2: Because:

> The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.

I think we can rewrite to something like this:

```suggestion
    We recommend you use the words hour(s), day(s), week(s), month(s) and year(s) in your time ranges.
```

}
```

### matchCurrentAge

Use this field if you want to match packages that satisfy the specified age range.

For example, if you want to group packages which are older than 2 months:

```json
{
  "packageRules": [
    {
      "matchCurrentAge": "> 2 months",
      "groupName": "vital"
    }
  ]
}
```

The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.

> Only _one_ date part is supported, so you _cannot_ do `> 1 year 1 month`.

Reviewer 2: ```suggestion

Only _one_ date part is supported, so you _cannot_ do `> 1 year 1 month`.
```

Might be good to separate the sentences with a newline.

Instead you should do `> 13 months`.

<!-- prettier-ignore -->
!!! note

>     We recommend you to only use the words hour(s), day(s), week(s), month(s) and year(s) in your time ranges.

Reviewer 2: ```suggestion
    We recommend you only use the words hour(s), day(s), week(s), month(s) and year(s) in your time ranges.
```

The `matchCurrentAge` string must start with one of `>`, `>=`, `<` or `<=`.
Only one date part is supported, e.g. you _cannot_ do `> 1 year 1 month` and instead should do `> 13 months`.

<!-- prettier-ignore -->
!!! note

>     Only the words hour(s)/hrs/h, day(s)/d, week(s), month(s)/M and year(s)/y are supported.

Reviewer 2: I think there's a typo somewhere. `Months` is `M`, but `Years` is `y`?

I would expect one of these:

1. Months is `M`, years is `Y`
2. Months is `m`, years is `y`
3. Or even: `M/Y` or `m/y`, casing does not matter

Reviewer 2: Let's use the full words in our docs and point the reader to the library's documentation if they want to know all the words.

Author: I left the link to the docs out because months is out addition and not supported by `ms`.

}
```

### matchCurrentAge

> Use this field if you want to match packages that satisfy the specified age range.

Reviewer: ```suggestion
Use this field if you want to match packages based on the age of the _current_ (existing, in-repo) version.
```

> For example, if you want to group packages which are older than 2 months:
> 
> ```json
> {
>   "packageRules": [
>     {
>       "matchCurrentAge": "> 2 months",
>       "groupName": "vital"
>     }
>   ]
> }

Reviewer: ```suggestion
For example, if you want to group updates for dependencies where the existing version is more than 2 years old:

```json
{
  "packageRules": [
    {
      "matchCurrentAge": "> 2 years",
      "groupName": "old dependencies"
    }
  ]
}
```
